### PR TITLE
[AIRFLOW-4487] Move k8s executor from contrib folder to main project

### PR DESCRIPTION
**Pull request by dimberman**
Considering that the k8s executor is now supported by core
committers, we should move it from contrib to the primary executor
directory.

Make sure you have checked _all_ steps below.

### Jira

- [ ] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [ ] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [ ] Passes `flake8`

### CODE REVIEW

**Review by feng-tao — APPROVED** [review]

**Review by kaxil — APPROVED** [review]

File: airflow/kubernetes/pod.py [hunk 1]
```diff
@@ -50,13 +50,13 @@ class Pod:
     :param secrets: Secrets to be launched to the pod
     :type secrets: list[airflow.contrib.kubernetes.secret.Secret]
     :param result: The result that will be returned to the operator after
-        successful execution of the pod
+    successful execution of the pod
```
Comment (feluelle) [inline airflow/kubernetes/pod.py]: I liked it more with indention, actually. Or was there in issue with that indention?

File: airflow/kubernetes/kubernetes_request_factory/pod_request_factory.py [hunk 2]
```diff
@@ -38,6 +39,7 @@ class SimplePodRequestFactory(KubernetesRequestFactory):
     """
 
     def __init__(self):
+
```
Comment (feluelle) [inline airflow/kubernetes/kubernetes_request_factory/pod_request_factory.py:42]: Looks strange to me, but okay :)

File: airflow/kubernetes/pod_launcher.py [hunk 3]
```diff
@@ -70,14 +69,18 @@ def delete_pod(self, pod):
...
+            pod: Pod,
+            startup_timeout: int = 120,
+            get_logs: bool = True) -> Tuple[State, Optional[str]]:
         """
         Launches the pod synchronously and waits for completion.
-        Args:
-            pod (Pod):
-            startup_timeout (int): Timeout for startup of the pod (if pod is pending for
-             too long, considers task a failure
+
```
Comment (feluelle) [inline airflow/kubernetes/pod_launcher.py:79]: Do you mind completing the documentation here? :)

**Review by feluelle — APPROVED** [review]
LGTM (only minor comments) 👍 

Is this graduation actually taking place in 1.10.4 or 2.0 ? I think 2.0 would be more appropriate. What do you guys think?

**Comment by kaxil** [comment]
Should go in 2.0 - would be a natural graduation :)

File: airflow/kubernetes/pod.py (hunk 1 above)
Comment (kaxil) [reply airflow/kubernetes/pod.py]: We might need indentation for Sphinx. Or else the params won't be rendered properly.

File: airflow/kubernetes/pod.py [hunk 4]
```diff
@@ -50,13 +50,13 @@ class Pod:
...
     :type result: any
     :param image_pull_policy: Specify a policy to cache or always pull an image
     :type image_pull_policy: str
     :param image_pull_secrets: Any image pull secrets to be given to the pod.
-        If more than one secret is required, provide a comma separated list:
-        secret_a,secret_b
+    If more than one secret is required, provide a comma separated list:
```
Comment (kaxil) [inline airflow/kubernetes/pod.py]: Same here.

File: airflow/kubernetes/pod.py [hunk 5]
```diff
@@ -67,8 +67,9 @@ class Pod:
     :param security_context: A dict containing the security context for the pod
     :type security_context: dict
     :param configmaps: A list containing names of configmaps object
-        mounting env variables to the pod
+    mounting env variables to the pod
```
Comment (kaxil) [inline airflow/kubernetes/pod.py]: and here

**Review by kaxil — APPROVED** [review]
Can we add some documentation as well on K8s executor as I feel we are light on it? - It can be a separate Jira issue and can be a part of another PR :)

**Comment by dimberman** [comment]
@kaxil I'm all for writing better docs on the k8sexecutor (which is part of why I'm starting to create these cleanup PRs). Is there any way we could start using epics in Jira? Would make this all a lot easier to track.

**Comment by kaxil** [comment]
We might have to probably raise Apache Infra ticket to create epic. But if for some reason if they deny that request - we can create a Parent Task (K8s cleanup) with many sub-tasks (Add docs, etc).

File: airflow/kubernetes/pod.py (hunk 1 above)
Comment (dimberman) [reply airflow/kubernetes/pod.py]: That was just me trying to figure out some sphinx errors. I can revert.

**Comment by ashb** [comment]
@dimberman Could you add a note about the rename in to UPDATING.md please?

**Pull request merged**